src/pages/home/report/PureReportActionItem.tsx

```diff
@@ -0,0 +1,1113 @@
+
+    /** Position index of the report action in the overall report FlatList view */
+    index: number;
+
+    /** Flag to show, hide the thread divider line */
+    shouldHideThreadDividerLine?: boolean;
+
+    linkedReportActionID?: string;
+
+    /** Callback to be called on onPress */
+    onPress?: () => void;
+
+    /** If this is the first visible report action */
+    isFirstVisibleReportAction: boolean;
+
+    /**
+     * Is the action a thread's parent reportAction viewed from within the thread report?
+     * It will be false if we're viewing the same parent report action from the report it belongs to rather than the thread.
+     */
+    isThreadReportParentAction?: boolean;
+
+    /** IF the thread divider line will be used */
+    shouldUseThreadDividerLine?: boolean;
+
+    /** Whether context menu should be displayed */
+    shouldDisplayContextMenu?: boolean;
+
+    /** ReportAction Draftmessage */
+    draftMessage?: string;
+
+    /** The IOU/Expense report we are paying */
+    iouReport?: OnyxTypes.Report;
+
+    /** All the emoji reactions for the report action. */
+    emojiReactions?: OnyxTypes.ReportActionReactions;
+
+    /** User's Expensify Wallet */
+    userWallet?: OnyxTypes.UserWallet;
+
+    /** Linked transaction route error */
+    linkedTransactionRouteError?: Errors;
+
+    /** Optional property for report name-value pairs */
+    reportNameValuePairs?: OnyxTypes.ReportNameValuePairs;
+
+    /** Optional property to indicate if the user is validated */
+    isUserValidated?: boolean;
+
+    /** Parent report */
+    parentReport?: OnyxTypes.Report;
+
+    /** Personal details list */
+    personalDetails?: OnyxTypes.PersonalDetailsList;
+
+    /** Whether or not the user is blocked from concierge */
+    blockedFromConcierge?: OnyxTypes.BlockedFromConcierge;
+};
+
+function PureReportActionItem({
+    action,
+    report,
+    transactionThreadReport,
+    linkedReportActionID,
+    displayAsGroup,
+    index,
+    isMostRecentIOUReportAction,
+    parentReportAction,
+    shouldDisplayNewMarker,
+    shouldHideThreadDividerLine = false,
+    shouldShowSubscriptAvatar = false,
+    onPress = undefined,
+    isFirstVisibleReportAction = false,
+    isThreadReportParentAction = false,
+    shouldUseThreadDividerLine = false,
+    shouldDisplayContextMenu = true,
+    parentReportActionForTransactionThread,
+    draftMessage,
+    iouReport,
+    emojiReactions,
+    userWallet,
+    linkedTransactionRouteError,
+    reportNameValuePairs,
+    isUserValidated,
+    parentReport,
+    personalDetails,
+    blockedFromConcierge,
+}: PureReportActionItemProps) {
+    const {translate} = useLocalize();
+    const {shouldUseNarrowLayout} = useResponsiveLayout();
+    const reportID = report?.reportID ?? '';
+    // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+    const originalReportID = useMemo(() => ReportUtils.getOriginalReportID(reportID, action) || '-1', [reportID, action]);
```

Comment:
`ReportUtils.getOriginalReportID` is still connected directly to Onyx which makes this component not really pure and will break when we use it in Search

Reply:
done